app-libs/stf/src/trusted_call.rs

```diff
@@ -197,7 +202,7 @@ impl ExecuteCall for TrustedCallSigned {
 				.map_err(|e| {
 					Self::Error::Dispatch(format!("Balance Set Balance error: {:?}", e.error))
 				})?;
-				Ok(())
+				Ok::<(), Self::Error>(())
```

Comment:
Dafuq, was the compiler having issues with type inference? Can't you leave it as it was before?

Reply:
It was (somehow), looks like the compiler can't infer the Error type even with https://github.com/integritee-network/worker/blob/15e388b406a20d420f252feec8079347aa7babba/app-libs/stf/src/trusted_call.rs#L235

and all conversion implementations: https://github.com/integritee-network/worker/blob/15e388b406a20d420f252feec8079347aa7babba/app-libs/stf/src/lib.rs#L79-L89

any better ideas?

```
error[E0282]: type annotations needed
   --> app-libs/stf/src/trusted_call.rs:205:5
    |
205 |                 Ok(())
    |                 ^^ cannot infer type of the type parameter `E` declared on the enum `Result`
    |
help: consider specifying the generic arguments
    |
205 |                 Ok::<(), E>(())
    |                   +++++++++

error[E0283]: type annotations needed
   --> app-libs/stf/src/trusted_call.rs:205:5
    |
205 |                 Ok(())
    |                 ^^ cannot infer type of the type parameter `E` declared on the enum `Result`
...
366 |         }?;
    |          - type must be known at this point
    |
note: multiple `impl`s satisfying `StfError: std::convert::From<_>` found
   --> app-libs/stf/src/lib.rs:79:1
    |
79  | impl From<MetadataError> for StfError {
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
...
85  | impl From<MetadataProviderError> for StfError {
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    = note: and another `impl` found in the `core` crate: `impl<T> std::convert::From<T> for T;`
    = note: required for `std::result::Result<(), StfError>` to implement `FromResidual<std::result::Result<std::convert::Infallible, _>>`
help: consider specifying the generic arguments
    |
205 |                 Ok::<(), E>(())
    |                   +++++++++
```

Reply:
Alright, I have taken a look at it, and I definitely see optimization that we can do, but it should not be in the scope of this PR, see: https://github.com/integritee-network/worker/issues/1145.

Reply:
The code does not look like it can be fixed easliy otherwise. The only thing we could do here is to replace the `?` here ` node_metadata_repo.get_from_metadata(|m| m.unshield_funds_call_indexes())??` with `map_err` so that we also tell the compiler explicitly what it needs to know.

The only thing I would ask you is to add a todo referencing #1145, this would be great. :)

Reply:
Definitely, thanks for looking into it. Yea, `map_err` doesn't make much of a difference than `Ok::<(), E>` :)

I'll add TODO and see if I can resolve #1145 in another PR

Reply:
added